src/esp/core/managedContainers/ManagedContainerBase.cpp

```diff
@@ -171,25 +171,43 @@ std::string ManagedContainerBase::getObjectInfoCSVString(
 std::string ManagedContainerBase::getUniqueHandleFromCandidatePerType(
     const std::map<int, std::string>& mapOfHandles,
     const std::string& name) const {
-  // find all existing values with passed name - this
-  std::vector<std::string> resVals =
+  // find all existing values with passed name - these should be a list
+  // of existing instances of this object name.  We are going to go through
+  // all of these names and find the "last" instance, meaning the highest count.
+  std::vector<std::string> objHandles =
       this->getObjectHandlesBySubStringPerType(mapOfHandles, name, true);
 
   int incr = 0;
-  // default to illegal apple/windows character
+  // use this as pivot character - the last instance of this character in any
+  // object instance name will be the partition between the name and the count.
   char pivotChar = ':';
-#if __linux__
-  pivotChar = '/';
-#endif
-  if (resVals.size() != 0) {
+
+  if (objHandles.size() != 0) {
     // handles exist with passed substring.  Find highest handle increment, add
     // 1 and use for new name 1, build new handle
-    for (const std::string& s : resVals) {
+    for (const std::string& objName : objHandles) {
+      // make sure pivot character is found in name
+      if (objName.find(pivotChar) == std::string::npos) {
+        continue;
```

Comment:
What's going on here? Why would this happen? (or can we just assert?)

Reply:
This function is also intended to be used by containers that do not normally manage numbered instances (such as for the attributes).  Intent is to allow for same-named configs to be auto-incremented before save, if the user doesn't wish to overwrite the existing config.

As far as the object wrappers/instances go (which is what you are interested in) this is superfluous - there will never be a handle that does not contain the pivot.